jerry-core/ecma/builtin-objects/ecma-builtin-json.cpp

```diff
@@ -597,34 +598,15 @@ ecma_builtin_json_parse_value (ecma_json_token_t *token_p) /**< token argument *
       bool parse_comma = false;
       uint32_t length = 0;
 
-#ifndef CONFIG_ECMA_COMPACT_PROFILE_DISABLE_ARRAY_BUILTIN
-      ecma_object_t *array_prototype_p = ecma_builtin_get (ECMA_BUILTIN_ID_ARRAY_PROTOTYPE);
-#else /* !CONFIG_ECMA_COMPACT_PROFILE_DISABLE_ARRAY_BUILTIN */
-      ecma_object_t *array_prototype_p = ecma_builtin_get (ECMA_BUILTIN_ID_OBJECT_PROTOTYPE);
-#endif /* CONFIG_ECMA_COMPACT_PROFILE_DISABLE_ARRAY_BUILTIN */
+      ecma_completion_value_t array_construction = ecma_op_create_array_object (NULL, 0, false);
+      JERRY_ASSERT (ecma_is_completion_value_normal (array_construction));
 
-      ecma_object_t *array_p = ecma_create_object (array_prototype_p, true, ECMA_OBJECT_TYPE_ARRAY);
-      ecma_deref_object (array_prototype_p);
-
-      ecma_property_t *class_prop_p = ecma_create_internal_property (array_p, ECMA_INTERNAL_PROPERTY_CLASS);
-      class_prop_p->u.internal_property.value = LIT_MAGIC_STRING_ARRAY_UL;
+      ecma_object_t *array_p = ecma_get_object_from_value (ecma_get_completion_value_value (array_construction));
 
       while (true)
       {
         if (ecma_builtin_json_check_right_square_token (token_p))
         {
-          ecma_string_t *length_magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_LENGTH);
-          ecma_number_t *length_num_p = ecma_alloc_number ();
-          *length_num_p = ecma_uint32_to_number (length);
-
-          ecma_property_t *length_prop_p = ecma_create_named_data_property (array_p,
-                                                                            length_magic_string_p,
-                                                                            true,
-                                                                            false,
-                                                                            false);
-          ecma_set_named_data_property_value (length_prop_p, ecma_make_number_value (length_num_p));
-          ecma_deref_ecma_string (length_magic_string_p);
-
```

Comment:
Tricky. There was no reference to ecma_op_array_object_define_own_property in the patch, so this is a hidden feature.